### PR TITLE
Add WAILA

### PR DESCRIPTION
**Pull request by bloxgate**
Adds WAILA support for all transparent nodes. Includes a configurable "Easy Mode" that shows more info on a few devices.

Also adds a plotBuckets and a readPrivateDouble function to Utils.

---

This change is [Reviewable](https://reviewable.io/reviews/electrical-age/electricalage/480)

### CODE REVIEW

**Comment by cm0x4D** [comment]
@bloxgate Is this ready to be merged? So it could be part of the next release...

**Comment by bloxgate** [comment]
@cm0x4D Yes. Last I checked it was working on all the machines, though I'd feel better if someone took a look and made sure I hadn't missed anything.

File: src/main/java/mods/eln/Eln.java [hunk 1]
```diff
@@ -297,11 +296,16 @@
...
 	double stdHalfLife = 2 * Utils.minecraftDay;
 
+	public static boolean wailaEasyMode = false;
+
 	@EventHandler
 	public void preInit(FMLPreInitializationEvent event) {
 		
-		achNetwork = NetworkRegistry.INSTANCE.newSimpleChannel("achChannel");
-		achNetwork.registerMessage(AchievePacketHandler.class, AchievePacket.class, 0, Side.SERVER);
+		elnNetwork = NetworkRegistry.INSTANCE.newSimpleChannel("eln");
```
Comment (cm0x4D) [inline src/main/java/mods/eln/Eln.java]: @bloxgate Renaming this channel has no side effects?

**Review by cm0x4D — COMMENTED** [review]
LGTM, I am just not sure that renaming the network channel does not have any unwanted side effects...

**Comment by Baughn** [comment]
Renaming the channel has the side-effect that you won't be able to connect to a server running this patch with a client that doesn't, or vice versa. That's not a big deal.

I'm concerned about possible name clashes, with it being just three letters, but I suppose it's still unlikely.

File: src/main/java/mods/eln/Eln.java (hunk 1 above)
Comment (cm0x4D) [reply src/main/java/mods/eln/Eln.java]: Maybe using "electricalage" or "electrical-age" in place of "eln"?

Comment (bloxgate) [reply src/main/java/mods/eln/Eln.java]: As mentioned below, as long as someone doesn't try using a different version than the server has it should be fine.

Comment (bloxgate) [reply src/main/java/mods/eln/Eln.java]: @cm0x4D I've changed the channel name to electrical-age

**Review by Baughn — APPROVED** [review]

**Comment by cm0x4D** [comment]
Reviewed 41 of 43 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/electrical-age/electricalage/480)_

**Comment by cm0x4D** [comment]
Merged, thanks @bloxgate for your contribution!

**Pull request merged**